src/Bicep.LangServer/Telemetry/BicepTelemetryEvent.cs

```diff
@@ -201,7 +201,50 @@ public static BicepTelemetryEvent CreateDeployStartOrWaitForCompletionResult(str
                 properties: new()
                 {
                     ["deployId"] = deployId,
-                    ["result"] = isSuccess ? Result.Succeeded : Result.Failed
+                    ["result"] = isSuccess ? EventResult.Succeeded : EventResult.Failed
+                }
+            );
+
+        public static BicepTelemetryEvent DecompileSuccess(string decompileId, int countOutputFiles, int countConflictingFiles)
+            => new BicepTelemetryEvent
+            (
+                eventName: TelemetryConstants.EventNames.DecompileSuccess,
+                properties: new()
+                {
+                    ["decompileId"] = decompileId.ToString(),
+                    ["countOutputFiles"] = countOutputFiles.ToString(),
+                    ["countConflictingFiles"] = countConflictingFiles.ToString(),
+                }
+            );
+
+        public static BicepTelemetryEvent DecompileFailure(string failureType)
+            => new BicepTelemetryEvent
+            (
+                eventName: TelemetryConstants.EventNames.DecompileFailure,
+                properties: new()
+                {
+                    ["failureType"] = failureType,
```

Comment:
Shouldn't we add  ["decompileId"] = decompileId.ToString() for this event?

Reply:
Thanks!